client/allochealth/tracker.go

```diff
@@ -191,10 +202,13 @@ func (t *Tracker) setTaskHealth(healthy, terminal bool) {
 func (t *Tracker) setCheckHealth(healthy bool) {
 	t.l.Lock()
 	defer t.l.Unlock()
-	t.checksHealthy = healthy
+
+	// check health should always be false if tasks are unhealthy
+	// as checks might be missing from unhealthy tasks
+	t.checksHealthy = healthy && t.tasksHealthy
```

Comment:
This calls out for having a state machine to represent these dependent states.  Would appreciate suggestions, specially ones that don't require significant rewrite.

Reply:
Your logic seems sound even if the structure of this package is sub-optimal. :+1: to your approach.